python-checks/src/main/java/org/sonar/python/checks/tests/NotDiscoverableTestMethodCheck.java

```diff
@@ -89,18 +125,21 @@ private static void checkSuspiciousFunctionsUsages(SubscriptionContext ctx, Map<
   }
 
   private static boolean inheritsOnlyFromUnitTest(ClassDef classDefinition) {
-    Optional<ClassSymbol> classSymbolFromDef = Optional.ofNullable(TreeUtils.getClassSymbolFromDef(classDefinition));
-    return classSymbolFromDef.filter(classSymbol -> !classSymbol.superClasses().isEmpty()).isPresent() &&
-      classSymbolFromDef
-      .map(ClassSymbol::superClasses)
-      .stream()
-      .flatMap(List::stream)
-      .map(Symbol::fullyQualifiedName)
-      .allMatch("unittest.case.TestCase"::equals);
+    return UnittestUtils.isInheritingFromUnittest(classDefinition) &&
+      Optional.ofNullable(TreeUtils.getClassSymbolFromDef(classDefinition)).stream()
+        .anyMatch(classSym -> classSym.superClasses().size() == 1);
```

Comment:
I see what you mean here, I tend to create Utils method even when I need them only once, just in case.
I guess this should be done the other way : we create this method locally in the class we need it, and in case we someday also need it somewhere else, we move it to a Utils class. The only difficulty I see here is that potentially we may not remember/be aware that this method already exist in another class and end up creating it in both place, but probably SonarQube is able to detect such duplication and warn us ?

Anyway, I will move this class into the class since it is only used here.

Reply:
SonarQube is pointing out duplicated code. However, there is always the risk to miss helper or already implemented logic.